src/commands/database-remove.js

```diff
@@ -38,32 +40,199 @@ module.exports = new Command("database:remove <path>")
+              return resolve();
+            });
+          });
+        });
+      }
+
+      // return whether it times out or not
+      function prefetchTest(path) {
+        var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + ".json?";
+        url += querystring.stringify({
+          timeout: "100ms",
+        });
+        var reqOptions = {
+          url: url,
+        };
+        return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+          return new Promise(function(resolve, reject) {
+            request.get(reqOptionsWithToken, function(err, res, body) {
+              if (err) {
+                return reject(
+                  new FirebaseError("Unexpected error while prefetching data to delete", {
+                    exit: 2,
+                  })
+                );
+              } else if (res.statusCode == 200) {
+                if (res) {
+                  return resolve("small");
+                } else {
+                  return resolve("empty");
+                }
+                // small subtree
+              } else if (res.statusCode == 400 || res.statusCode == 413) {
+                // timeout or payload too large
+                // large subtree, recursive delete for each subtree
+                return resolve("larger");
+              } else {
+                return reject(responseToError(res, body));
+              }
+            });
+          });
+        });
+      }
+
+      function listPath(path) {
+        var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + ".json?";
+        if (path === "/") {
+          url += querystring.stringify({
+            shallow: true,
+          });
+        } else {
+          url += querystring.stringify({
+            shallow: true,
+            limitToFirst: "100",
+          });
+        }
+        var reqOptions = {
+          url: url,
+        };
+        return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+          return new Promise(function(resolve, reject) {
+            request.get(reqOptionsWithToken, function(err, res, body) {
+              if (err) {
+                return reject(
+                  new FirebaseError("Unexpected error while list subtrees", {
+                    exit: 2,
+                  })
+                );
+              } else if (res.statusCode >= 400) {
+                return reject(responseToError(res, body));
+              }
+              var data = {};
+              try {
+                data = JSON.parse(body);
+              } catch (e) {
+                return reject(
+                  new FirebaseError("Malformed JSON response in shallow get ", {
+                    exit: 2,
+                    original: e,
+                  })
+                );
+              }
+              if (data) {
+                var keyList = Object.keys(data);
+                return resolve(keyList);
+              } else {
```

Comment:
else is unnecessary. Since it returns in the `if` above, the else can be removed and simply return after the `if` statement